### PR TITLE
ramips: add support for TOTOLINK A7000R

### PR DESCRIPTION
**Pull request by 981213**
This PR adds support for TOTOLINK A7000R and also fixed gpio interrupt support for mt7621.
The fix for gpio interrupt could be backported to 19.07 branch.

### CODE REVIEW

**Comment by ynezz** [comment]
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

**Comment by ynezz** [comment]
I'll leave it open so we don't forget to backport that IRQ fix into 19.07.

File: target/linux/ramips/dts/TOTOLINK-A7000R.dts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@14c3,7615 {
```
Comment (chunkeey) [inline target/linux/ramips/dts/TOTOLINK-A7000R.dts]: it's wifi@0,0 (The unit address is taken from the reg value)

File: target/linux/ramips/dts/TOTOLINK-A7000R.dts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		compatible = "pci14c3,7615";
```
Comment (chunkeey) [inline target/linux/ramips/dts/TOTOLINK-A7000R.dts]: The binding document example also specifies "mediatek,mt76". 
https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/net/wireless/mediatek%2Cmt76.txt

But either way is OK.

File: target/linux/ramips/dts/TOTOLINK-A7000R.dts (hunk 1 above)
Comment (981213) [reply target/linux/ramips/dts/TOTOLINK-A7000R.dts]: Oops. Fixed.
@ynezz Could you update my commit in your staging tree?

Comment (ynezz) [reply target/linux/ramips/dts/TOTOLINK-A7000R.dts]: Done, thanks.

**Comment by ynezz** [comment]
That IRQ fix is now in 19.07 so closing this PR.

**Pull request closed without merging**